app/bundles/EmailBundle/Controller/AjaxController.php

```diff
@@ -285,4 +286,56 @@ public function getEmailDeliveredCountAction(Request $request, CacheProvider $ca
             'delivered'   => $deliveredCount,
         ]);
     }
+
+    public function heatmapAction(Request $request): JsonResponse
+    {
+        $emailId = (int) InputHelper::clean($request->query->get('id'));
+
+        /** @var EmailModel $model */
+        $model = $this->getModel('email');
+
+        $email       = $model->getEntity($emailId);
+
+        if (null === $email) {
+            return $this->sendJsonResponse([
+                'success' => 0,
+                'message' => $this->translator->trans('mautic.api.call.notfound'),
+            ], 404);
+        }
+
+        if (!$this->security->hasEntityAccess(
+            'email:emails:viewown',
+            'email:emails:viewother',
+            $email->getCreatedBy()
+        )
+        ) {
+            return $this->accessDenied();
+        }
+
+        $content   = $email->getCustomHtml();
+        $content   = EmojiHelper::toEmoji($content, 'short');
```

Comment:
I'd like to get rid of the EmojiHelper in Mautic 6. It used to be necessary as the UTF8 Mysql encoding couldn't work with some emojis that needed UTF16 space. But all Mautic instances created after Mautic 3 release are on UTF8-MB4 encoding that have no issue with emojis. So I think it's time to sunset the EmojiHelper.

Reply:
Thanks for clarification, I will remove this